src/inverse-kinematics/include/iDynTree/InverseKinematics.h

```diff
@@ -166,6 +167,84 @@ class iDynTree::InverseKinematics
 
     enum InverseKinematicsRotationParametrization rotationParametrization();
 
+    /*! @name Parameters-related methods
+     */
+    ///@{
+
+    /**
+     * Sets Maximum Iteration.
+     *
+     * The default value for this parameter is 3000 .
+     *
+     * @param max_iter exits if iter>=max_iter (max_iter<0
+     *                   disables this check).
+     */
+    void setMaxIter(const int max_iter);
```

Comment:
For the options I would have liked a sort of single method with key-value pairs, but I think that given the time constraints we can live it in this way